File: src/main/java/com/example/demo/model/Bicycle.java
package com.example.demo.model;

import jakarta.persistence.*;

@Entity
public class Bicycle {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    private String serialNumber;
    private String brand;
    private String model;

    @ManyToOne
    @JoinColumn(name = "owner_id")
    private Owner owner;

    // =========================
    // GETTERS & SETTERS
    // =========================

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getSerialNumber() {
        return serialNumber;
    }

    public void setSerialNumber(String serialNumber) {
        this.serialNumber = serialNumber;
    }

    public String getBrand() {
        return brand;
    }

    public void setBrand(String brand) {
        this.brand = brand;
    }

    public String getModel() {
        return model;
    }

    public void setModel(String model) {
        this.model = model;
    }

    public Owner getOwner() {
        return owner;
    }

    public void setOwner(Owner owner) {
        this.owner = owner;
    }
}

File: src/main/java/com/example/demo/model/Bicycle.kt
package com.example.demo.model

import jakarta.persistence.*

@Entity
class Bicycle {
    // =========================
    // GETTERS & SETTERS
    // =========================
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long = 0

    var serialNumber: String? = null
    var brand: String? = null
    var model: String? = null

    @ManyToOne
    @JoinColumn(name = "owner_id")
    var owner: Owner? = null
}